src/Common/FrequencyHolder.h

```diff
@@ -61,6 +62,35 @@ class FrequencyHolder
         return instance;
     }
 
+    const Map & getEmotionalDict()
+    {
+        std::lock_guard lock(mutex);
+        if (emotional_dict.empty())
+            loadEmotionalDict();
+
+        return emotional_dict;
```

Comment:
I know it's not part of the PR, but this doesn't seem like a good practice
We are exposing internal object protected by `mutex`.

```
static auto emotional_dict = std::invoke([this] {
    return loadEmotionalDict();
};
return emotional_dict
```
should work and be thread-safe (if you modify the `load` function to return a locally created emotional dict.

Reply:
Not sure if I understand this comment. First
```c++
static auto emotional_dict = std::invoke([this] {
    return loadEmotionalDict();
};
```
This construction will use mutex, and atomic underneath.
Current implementation is not as efficient, we can make it more efficient if we protect each dictionary with atomic and mutex, but probably this could be addressed in separate pull request.

Reply:
The problem is from the API standpoint not the performance. 
With the above construct, you ensure that the object is created only once in a thread-safe way and from that point it cannot be modified from any other function so sending the ref to it is safe.

With the current implementation, you're returning a reference to an object that is internally protected by mutex.
In theory, you can get the reference to the object, and while reading it, the object can be modified from another thread because the mutex is not locked.
This won't happen because of the limited way the objects are used in the current implementation, but it's not hard to imagine that it can happen in a more complex scenario. 

This was more of an advice comment for general case so I'm 100% okay with leaving this as is but it's a good thing to keep in mind. (`C++ Concurrency in Action` describes cases like that in good detail if you're interested).

Reply:
Thanks for explanation. Make sense.